ee/desktop/server/server.go

```diff
@@ -0,0 +1,71 @@
+package server
+
+import (
+	"context"
+	"net/http"
+	"os"
+
+	"fyne.io/systray"
+	"github.com/kolide/launcher/ee/desktop"
+)
+
+var shutdownChan = make(chan struct{})
+
+// Start spins up a server that listens to a unix socket or named pipe for windows.
+//
+// On macos & linux you can curl this with:
+//
+//	curl --unix-socket /tmp/<pid>_launcher_desktop.sock ./<endpoint>
+//
+// For example:
+//
+//	curl --unix-socket /tmp/1234_launcher_desktop.sock ./shutdown
+//
+// On windows the socket path is \\.\pipe\<pid>_launcher_desktop.sock, but curl
+// doesn't seem to support named pipes on windows.
+func Start() error {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/shutdown", shutdown)
+
+	server := http.Server{
+		Handler: mux,
+	}
+
+	socketPath := desktop.DesktopSocketPath(os.Getpid())
+
+	if err := os.RemoveAll(socketPath); err != nil {
```

Comment:
Feels weird. Might be correct.

Reply:
I'm not 100% sure either, but it feels simplest to just let the dialer create the socket